catalog/transformer/catalog-transformer-csv-queryresponse/src/test/java/ddf/catalog/transformer/csv/CsvMetacardTransformerTest.java

```diff
@@ -0,0 +1,159 @@
+
+    private static Set<AttributeDescriptor> attributeDescriptorsSet;
+
+    private static List<AttributeDescriptor> attributeDescriptorsList;
+
+    private static Map<String, Attribute> metacardDataMap;
+
+    private static List<AttributeDescriptor> ATTRIBUTE_DESCRIPTOR_LIST;
+
+    Map<String, Serializable> arguments;
+
+    private CsvMetacardTransformer transformer;
+
+    Metacard normalMC;
+
+    private static final List<ImmutableTriple<Object, Object, Object>> ATTRIBUTE_DATA =
+            Arrays.asList(
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute1", "value1", BasicTypes.STRING_TYPE),
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute2", "value2", BasicTypes.STRING_TYPE),
+                    new ImmutableTriple<Object, Object, Object>("attribute3", 101, BasicTypes.INTEGER_TYPE),
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute4", 3.14159, BasicTypes.DOUBLE_TYPE),
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute5", "value,5", BasicTypes.STRING_TYPE),
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute6", "value6", BasicTypes.STRING_TYPE),
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute7", "OBJECT", BasicTypes.OBJECT_TYPE),
+                    new ImmutableTriple<Object, Object, Object>(
+                            "attribute8", "BINARY", BasicTypes.BINARY_TYPE));
+
+    @Before
+    public void setUp() {
+        this.transformer = new CsvMetacardTransformer();
+        this.arguments = new HashMap<>();
+        attributeDescriptorsSet = new HashSet<>();
+        attributeDescriptorsList = new ArrayList<>();
+        metacardDataMap = new HashMap<>();
+        ATTRIBUTE_DESCRIPTOR_LIST = new ArrayList<>();
+
+        String[] hiddenFieldsArray = {"attribute3", "attribute5"};
+        arguments.put("hiddenFields", buildSet(hiddenFieldsArray));
+        String[] columnOrderArray = {"attribute3", "attribute4", "attribute2"};
+        arguments.put("columnOrder", buildList(columnOrderArray));
+        String[][] aliases = {{"attribute1", "column1"}, {"attribute2", "column2"}};
+        arguments.put("aliases", buildMap(aliases));
+
+        attributeDescriptorsSet.add(buildAttributeDescriptor("attribute1", BasicTypes.STRING_TYPE));
+        attributeDescriptorsSet.add(buildAttributeDescriptor("attribute2", BasicTypes.STRING_TYPE));
+        attributeDescriptorsSet.add(buildAttributeDescriptor("attribute3", BasicTypes.STRING_TYPE));
+        attributeDescriptorsSet.add(buildAttributeDescriptor("1", BasicTypes.INTEGER_TYPE));
+
+        buildMetacardDataMap();
+        normalMC = buildMetacard();
+
+    }
+
+    @Test
+    public void testCsvMetacardTransformer() throws CatalogTransformerException {
```

Comment:
✏️ The method's name should describe what it does. So something like `testCsvMetacardTransformerMimeType`